guides/NEPTUNE.md

```diff
@@ -0,0 +1,101 @@
+Migrating from Neo4j to AWS Neptune
+=======================
+
+A developer guide for moving the CMD services from using Neo4j to AWS Neptune.
+
+### Introduction to Neptune
+
+- Neptune is a graph database that is provided as a managed AWS service. This saves a lot of the operational overhead we had with Neo4j, and provides clustering capabilities out of the box.
+- Neptune supports both Gremlin and SPARQL as graph query languages. Our implementation is using Gremlin.
+- A Neptune cluster has a primary instance for read and write operations, and up to 15 read replica instances. When connecting to Neptune, the cluster endpoint will connect to the primary instance. There is also a reader endpoint which will connect to one of the available read only instances.
+- Neptune mostly complies with Gremlin standards but there are some caveats. This is the reason we cannot simply run another Gremlin compliant graph database locally. [See here](https://docs.aws.amazon.com/neptune/latest/userguide/feature-overview-standards-compliance.html) for more details on the standard compliance.
+- For more information on Neptune refer to the [Neptune docs](https://docs.aws.amazon.com/neptune/latest/userguide/intro.html)
+
+### Local development - How do I run the CMD services against Neptune?
+
+Neo4j runs within a docker container as part of the [dp-compose](https://github.com/ONSdigital/dp-compose) project. Running a local instance of Neptune is not possible, so we must use a test instance hosted in AWS.
+
+#### Use SSH port forwarding to connect to a Neptune cluster
+
+First you need the address of the cluster you want to connect to. You can get this from the AWS console:
+
+1. Go to the Neptune section of the AWS console https://eu-west-1.console.aws.amazon.com/neptune
+1. Select Databases from the side menu (it may already be selected)
+1. Click on the link for the cluster you want to connect to
+
+    ![1](../img/neptune_db_select.png)
+
+1. The cluster address can now be copied from the 'Connectivity & security' section at the bottom
+    
+    ![1](../img/neptune_cluster_address.png)
+
+1. In the terminal, CD into the ansible directory of the dp-setup project:
+```
+cd dp-setup/ansible
+```
+1. Run the command to port forward to the Neptune cluster:
+```
+ssh -F ssh.cfg -L 8182:{cluster address}:8182 {user name}@{publishing asg node IP}
```

Comment:
we may want to add the dp tool alternative, to do ssh? Would it work?
```
dp ssh -p 8182:{cluster address}:22521 develop publishing {num}
```
where num corresponds to the node number with the as node IP that you want

Reply:
Good point. I had it on my todo to investigate a custom dp tool action to connect to Neptune, however this looks like a better solution for the short term. I will give it a go 👍